### PR TITLE
Add support for DuckDB 1.1.0

### PR DESCRIPTION
**Pull request by stenlarsson**
This fixes a number of errors due to changes in DuckDB and ruby-duckdb between 0.5.0 and 1.1.0.

### CODE REVIEW

**Comment by stenlarsson** [comment]
To build the gem on my macOS machine I had to specify the duckdb dir. Not sure why it is not found automatically?
```
CONFIGURE_ARGS=--with-duckdb-dir=$(brew --prefix duckdb) rake install
```

Also, the GitHub Actions workflow no longer works.

**Comment by kou** [comment]
Thanks.
I'll fix our CI before I review this. Please wait for a white...

**Comment by kou** [comment]
Could you rebase on master?

**Comment by stenlarsson** [comment]
Seems to work now, thanks! https://github.com/stenlarsson/red-arrow-duckdb/actions/runs/11249869803

File: ext/arrow-duckdb/arrow-duckdb.cpp [hunk 1]
```diff
@@ -99,7 +101,7 @@ namespace {
   result_ensure_gschema(Result *result)
   {
     ArrowSchema c_abi_schema;
-    duckdb_arrow_schema schema = &c_abi_schema;
+    duckdb_arrow_schema schema = reinterpret_cast<duckdb_arrow_schema>(&c_abi_schema);
```
Comment (kou) [inline ext/arrow-duckdb/arrow-duckdb.cpp]:
```suggestion
    auto schema = reinterpret_cast<duckdb_arrow_schema>(&c_abi_schema);
```

File: ext/arrow-duckdb/arrow-duckdb.cpp [hunk 2]
```diff
@@ -121,7 +123,7 @@ namespace {
   result_fetch_internal(VALUE self, Result *result)
   {
     ArrowArray c_abi_array = {};
-    duckdb_arrow_array array = &c_abi_array;
+    duckdb_arrow_array array = reinterpret_cast<duckdb_arrow_array>(&c_abi_array);
```
Comment (kou) [inline ext/arrow-duckdb/arrow-duckdb.cpp]:
```suggestion
    auto array = reinterpret_cast<duckdb_arrow_array>(&c_abi_array);
```

File: ext/arrow-duckdb/arrow-duckdb.cpp [hunk 3]
```diff
@@ -220,7 +222,7 @@ namespace {
   query_sql_arrow(VALUE self, VALUE sql)
   {
     rubyDuckDBConnection *ctx;
-    Data_Get_Struct(self, rubyDuckDBConnection, ctx);
+    ctx = get_struct_connection(self);
```
Comment (kou) [inline ext/arrow-duckdb/arrow-duckdb.cpp]: Could you remove `rubyDuckDBConnection *ctx;` and use `auto ctx = ...` instead?

**Review by kou — COMMENTED** [review]
Thanks!

**Comment by stenlarsson** [comment]
Updated according to review comments.

**Comment by kou** [comment]
Thanks!

**Pull request merged**